### PR TITLE
candy-icons: init at unstable-2023-12-31

### PR DESCRIPTION
**Pull request by clr-cera**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/data/icons/candy-icons/default.nix
Comment (donovanglover) [inline pkgs/data/icons/candy-icons/default.nix:5]:
```suggestion
{ lib
, stdenv
, fetchFromGitHub
}:
```
Small nit since this formatting is more popular in nixpkgs

File: pkgs/data/icons/candy-icons/default.nix
Comment (donovanglover) [inline pkgs/data/icons/candy-icons/default.nix:9]:
```suggestion
  version = "unstable-2023-10-03";
```
Upstream doesn't have a release tagged with the date. Make sure to edit the commit message and PR title as well

File: pkgs/data/icons/candy-icons/default.nix
Comment (donovanglover) [inline pkgs/data/icons/candy-icons/default.nix:15]:
```suggestion
    hash = "sha256-NCXxH04GvrPE5mQO7JrU6SkjA+BaZokhyjKsm8m9p3g=";
```
Prefer `hash` over `sha256`

File: pkgs/data/icons/candy-icons/default.nix
Comment (donovanglover) [inline pkgs/data/icons/candy-icons/default.nix:21]:
```suggestion
  installPhase = ''
    runHook preInstall
    
    mkdir -p $out/share/icons/candy-icons
    cp -r . $out/share/icons/candy-icons
    
    runHook postInstall
  '';
```
See: https://nixos.org/manual/nixpkgs/unstable/#sec-stdenv-phases

File: pkgs/top-level/all-packages.nix
Comment (donovanglover) [inline pkgs/top-level/all-packages.nix:29290]: Use the new [`pkgs/by-name`](https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name) format so you don't have to edit this file

**Review by donovanglover — COMMENTED** [review]
Thanks for contributing to nixpkgs!

File: pkgs/top-level/all-packages.nix
Comment (clr-cera) [reply pkgs/top-level/all-packages.nix]: Thank you for your review! I have made all changes.

**Comment by clr-cera** [comment]
> Thanks for contributing to nixpkgs!

Thank you! I have resolved all suggested changes, if there is anything else to solve please let me know!

File: pkgs/by-name/ca/candy-icons/package.nix
Comment (donovanglover) [inline pkgs/by-name/ca/candy-icons/package.nix:2]:
```suggestion
, stdenvNoCC
```
NoCC since we're not compiling anything

File: pkgs/by-name/ca/candy-icons/package.nix
Comment (donovanglover) [inline pkgs/by-name/ca/candy-icons/package.nix:6]:
```suggestion
stdenvNoCC.mkDerivation {
```

File: pkgs/by-name/ca/candy-icons/package.nix
Comment (donovanglover) [inline pkgs/by-name/ca/candy-icons/package.nix:16]:
```suggestion

  nativeBuildInputs = [
    gtk3
  ];

  dontDropIconThemeCache = true;

```
See https://nixos.org/manual/nixpkgs/stable/#ssec-gnome-packaging

File: pkgs/by-name/ca/candy-icons/package.nix
Comment (donovanglover) [inline pkgs/by-name/ca/candy-icons/package.nix:4]:
```suggestion
, gtk3
}:
```

File: pkgs/by-name/ca/candy-icons/package.nix
Comment (donovanglover) [inline pkgs/by-name/ca/candy-icons/package.nix:21]:
```suggestion
    cp -r . $out/share/icons/candy-icons
    gtk-update-icon-cache $out/share/icons/candy-icons
```

**Review by donovanglover — COMMENTED** [review]
Make sure to squash the commits so there's 1 for the maintainer and 1 for the package.

The icons work for me so it should be good to go after this. :+1:

**Comment by clr-cera** [comment]
Hi! I have just accepted the suggestions and organized all commits. I am very thankful for all the help!

**Comment by donovanglover** [comment]
Result of `nixpkgs-review pr 259294` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>candy-icons</li>
  </ul>
</details>

**Review by donovanglover — APPROVED** [review]
LGTM

**Comment by D3vil0p3r** [comment]
@clr-cera can you refer the source to the latest commit please?

File: pkgs/by-name/ca/candy-icons/package.nix
Comment (D3vil0p3r) [inline pkgs/by-name/ca/candy-icons/package.nix:9]:
```suggestion
  version = "unstable-2023-12-31";
```
Changed to the latest commit date.

File: pkgs/by-name/ca/candy-icons/package.nix
Comment (D3vil0p3r) [inline pkgs/by-name/ca/candy-icons/package.nix:14]:
```suggestion
    rev = "e4464d7b4d8e1821025447b2064b6a8f5c4c8c89";
```
Changed to the latest commit.

File: pkgs/by-name/ca/candy-icons/package.nix
Comment (D3vil0p3r) [inline pkgs/by-name/ca/candy-icons/package.nix:15]:
```suggestion
    hash = "sha256-XdYjxWf8R4b1GK2iFQnoEOWykc19ZT37ki83WeESQBM=";
```

**Review by D3vil0p3r — CHANGES_REQUESTED** [review]
Update to the latest commit

**Review by eclairevoyant — CHANGES_REQUESTED** [review]
Thanks for the contribution! Also, please change the first commit's message to `maintainers: add clr-cera`

**Comment by D3vil0p3r** [comment]
@clr-cera thanks. Can you squash [changing source info to latest commit](https://github.com/NixOS/nixpkgs/pull/259294/commits/a2e1bd87e39d45e065658569dd09183260d51ac8) commit and change the commit message of [candy-icons: init at unstable-2023-10-03](https://github.com/NixOS/nixpkgs/pull/259294/commits/3a3007fe1e189904b670f84abfbfd4822339f7aa) to "candy-icons: init at unstable-2023-12-31"?

File: pkgs/by-name/ca/candy-icons/package.nix
Comment (D3vil0p3r) [inline pkgs/by-name/ca/candy-icons/package.nix:38]:
```suggestion
    license = licenses.gpl3Plus;
```
Last change

**Review by D3vil0p3r — CHANGES_REQUESTED** [review]

**Review by D3vil0p3r — APPROVED** [review]
LGTM

**Comment by clr-cera** [comment]
Thank you for your advice! All suggested changes have been committed and I hope this package can be merged!

File: pkgs/by-name/ca/candy-icons/package.nix
Comment (AndersonTorres) [inline pkgs/by-name/ca/candy-icons/package.nix:37]:
```suggestion
    description = "An icon theme colored with sweet gradients";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by eclairevoyant** [comment]
Apologies, the closure was accidental.
Can the PR title be updated to the right version?

**Comment by clr-cera** [comment]
> Apologies, the closure was accidental.
> Can the PR title be updated to the right version?
> 

Sure!

**Review by eclairevoyant — APPROVED** [review]

**Comment by D3vil0p3r** [comment]
@clr-cera can you apply the proposal made by @AndersonTorres ?

**Comment by D3vil0p3r** [comment]
Thank you @clr-cera for the update. Can you also squash the last commit?

**Review by h7x4 — APPROVED** [review]

**Comment by D3vil0p3r** [comment]
@h7x4 what is it needed to merge this PR?

**Pull request merged**
